src/fluree/db/storage/file.cljc

```diff
@@ -51,32 +73,61 @@
                         {:root root
                          :path dir
                          :data data})))
-      (let [hash     (crypto/sha2-256 data :base32)
-            filename (str hash ".json")
-            path     (str/join "/" [dir filename])
-            absolute (full-path root path)
-            bytes    (if (string? data)
-                       (bytes/string->UTF8 data)
-                       data)]
+      (let [hash          (crypto/sha2-256 data :base32)
+            filename      (str hash ".json")
+            path          (str/join "/" [dir filename])
+            absolute      (full-path root path)
+            original-size (count (if (string? data)
+                                   (bytes/string->UTF8 data)
+                                   data))
+            bytes         (cond-> data
+                            (string? data) bytes/string->UTF8
+                            encryption-key (aes/encrypt encryption-key {:output-format :none}))]
         (<? (fs/write-file absolute bytes))
         {:path    path
          :address (file-address identifier path)
          :hash    hash
-         :size    (count bytes)})))
+         :size    original-size})))
 
   storage/ByteStore
   (write-bytes [_ path bytes]
-    (-> root
-        (full-path path)
-        (fs/write-file bytes)))
+    (let [final-bytes (if encryption-key
+                        (aes/encrypt bytes encryption-key {:output-format :none})
+                        bytes)]
+      (-> root
+          (full-path path)
+          (fs/write-file final-bytes))))
 
   (read-bytes [_ path]
-    (-> root
-        (full-path path)
-        fs/read-file)))
+    (go-try
+      #?(:clj
+         ;; For Clojure, we need to read raw bytes
+         (let [file-path (full-path root path)]
+           (try
+             (let [file (java.io.File. file-path)]
+               (when (.exists file)
+                 (let [raw-bytes (java.nio.file.Files/readAllBytes (.toPath file))]
```

Comment:
Instead of this deeply nested structure, I think this code could be improved with a few file utility functions.
- something like `(file-exists? path)` could take a string path and report if a file exists at that path.
- something like `(read-bytes path)` could take a string path and read the bytes stored there.
- a third function combining the two that automatically returns `nil` if no file exists at that path, and the stored bytes if the file does exist.

Reply:
I refactored it a bit to keep the file storage protocol implementation clean and move the conditional logic into the util file which I think addresses your concern and feels more reasonable